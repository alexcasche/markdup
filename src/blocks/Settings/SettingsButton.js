import { ButtonLink } from 'elements'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { shade, tint } from 'polished'

const MODIFIERS = {
  disabled: ({ theme }) => `
    border-radius: 4px 0 0 4px;
    pointer-events: none;
    opacity: 0.85;
  `,
}

const SettingsButton = ButtonLink.extend`
  width: 50%;
  margin-top: 15px;
  height: 40px;
  background-color: ${props => shade(0.75, props.theme.colors.blue)};
  border-radius: 0 4px 4px 0;
  color: ${props => tint(0.15, props.theme.colors.grey)};
  font-size: 14px;
  &:hover {
    background-color: ${props => shade(0.85, props.theme.colors.blue)};
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default SettingsButton
