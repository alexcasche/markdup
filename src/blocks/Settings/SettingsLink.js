import { Link } from 'elements'
import { shade, tint } from 'polished'

const SettingsLink = Link.extend`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: ${props => props.theme.colors.grey};
  height: 50px;
  text-transform: capitalize;
  border-bottom: 1px solid ${props => shade(0.5, props.theme.colors.grey)};
  transition: color 0.15s ease-in-out;
  &:hover {
    color: ${props => tint(0.25, props.theme.colors.grey)};
    cursor: pointer;
  }
  &:last-of-type {
    border-bottom: none;
  }
`

export default SettingsLink
