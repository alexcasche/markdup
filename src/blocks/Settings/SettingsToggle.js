import { Toggle } from 'elements'
import { applyStyleModifiers } from 'styled-components-modifiers'
import { shade } from 'polished'
import 'react-toggle/style.css'

const MODIFIERS = {
  checked: ({ theme }) => `
    .react-toggle-track {
      background-color: ${shade(0.9, theme.colors.green)} !important;
    }
  `,
}

const SettingsToggle = Toggle.extend`
  pointer-events: none;
  .react-toggle-track {
    background-color: ${props => shade(0.5, props.theme.colors.grey)};
  }
  ${applyStyleModifiers(MODIFIERS)};
`

export default SettingsToggle
