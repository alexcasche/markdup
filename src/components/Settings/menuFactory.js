import React from 'react'
import { Settings } from 'blocks'

const menuFactory = ({ settingsLinks, toggleSetting }) => {
  return Object.keys(settingsLinks).map((link, index) => {
    const toggleModifiers = [...(settingsLinks[link] ? ['checked'] : [])]
    const text = link
      .split(/(?=[A-Z])/)
      .join(' ')
      .toLowerCase()
    return (
      <Settings.Link key={index} onClick={() => toggleSetting(link)}>
        {text}
        <Settings.Toggle
          checked={settingsLinks[link]}
          icons={false}
          readOnly={true}
          modifiers={toggleModifiers}
        />
      </Settings.Link>
    )
  })
}

export default menuFactory
