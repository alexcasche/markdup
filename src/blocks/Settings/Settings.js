import styled from 'styled-components'

import SettingsButton from './SettingsButton'
import SettingsLink from './SettingsLink'
import SettingsToggle from './SettingsToggle'

const Settings = styled.nav`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  z-index: 1;
  position: fixed;
  top: 70px;
  right: 30px;
  max-height: calc(100vh - 70px);
  overflow-y: scroll;
  background-color: ${props => props.theme.colors.black};
  border-radius: 4px;
  padding: 12px 20px 30px;
  width: 300px;
`

Settings.Button = SettingsButton
Settings.Link = SettingsLink
Settings.Toggle = SettingsToggle

export default Settings
