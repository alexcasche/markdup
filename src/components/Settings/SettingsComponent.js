import React from 'react'
import { Settings } from 'blocks'
import { default as Menu } from './menuFactory'
import { PropTypes as t } from 'prop-types'

Settings.Menu = Menu

const SettingsComponent = props => {
  const { showSettings, settingsLinks, actions } = props
  if (showSettings) {
    return (
      <Settings className="Settings">
        <Settings.Menu
          settingsLinks={settingsLinks}
          toggleSetting={actions.toggleSetting}
        />
        <Settings.Link
          onClick={() => actions.updateLayout({ showDialog: 'resetSession' })}
        >
          reset session
        </Settings.Link>
        <Settings.Button target="_blank" href="./" modifiers={['disabled']}>
          markdup
        </Settings.Button>
        <Settings.Button target="_blank" href="'./">
          styldup
        </Settings.Button>
      </Settings>
    )
  } else return null
}

SettingsComponent.propTypes = {
  showSettings: t.bool.isRequired,
  settingsLinks: t.shape({
    autoSave: t.bool.isRequired,
    liveComplete: t.bool.isRequired,
    splitView: t.bool.isRequired,
    syncEditors: t.bool.isRequired,
  }).isRequired,
  actions: t.shape({
    toggleSetting: t.func.isRequired,
    updateLayout: t.func.isRequired,
  }).isRequired,
}

export default SettingsComponent
