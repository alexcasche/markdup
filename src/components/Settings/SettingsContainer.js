import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getLayoutVal, getSettingVal } from 'selectors'
import { toggleSetting, updateLayout } from 'actions'
import SettingsComponent from './SettingsComponent'

export const SettingsContainer = props => {
  return <SettingsComponent {...props} />
}

function mapStateToProps(state) {
  return {
    settingsLinks: {
      autoSave: getSettingVal(state, 'autoSave'),
      liveComplete: getSettingVal(state, 'liveComplete'),
      splitView: getSettingVal(state, 'splitView'),
      syncEditors: getSettingVal(state, 'syncEditors'),
    },
    showSettings: getLayoutVal(state, 'showSettings'),
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: {
      toggleSetting: bindActionCreators(toggleSetting, dispatch),
      updateLayout: bindActionCreators(updateLayout, dispatch),
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SettingsContainer)
