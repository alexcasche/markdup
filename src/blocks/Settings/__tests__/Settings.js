import Settings from '../Settings'

describe('Settings Block', () => {
  it('renders correctly', () => {
    const tree = render(<Settings />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
