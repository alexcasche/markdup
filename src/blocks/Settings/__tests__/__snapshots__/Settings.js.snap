// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Block renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  z-index: 1;
  position: fixed;
  top: 70px;
  right: 30px;
  max-height: calc(100vh - 70px);
  overflow-y: scroll;
  background-color: #1D1F21;
  border-radius: 4px;
  padding: 12px 20px 30px;
  width: 300px;
}

<nav
  className="c0"
/>
`;
